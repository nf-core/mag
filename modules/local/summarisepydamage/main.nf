process SUMMARISEPYDAMAGE {
    tag "${meta.id}"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/pydamage:1.0--pyhdfd78af_0'
        : 'biocontainers/pydamage:1.0--pyhdfd78af_0'}"

    input:
    tuple val(meta), path(csv)

    output:
    tuple val(meta), path("*.tsv"), emit: summary_tsv
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    summarise_pydamage.py \\
        ${args} \\
        -i ${csv} \\
        -o ${prefix}_pydamagebins_summarised.tsv \\
        -n ${meta.id}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        summarisepydamage: \$(summarise_pydamage.py --version | cut -d ' ' -f 2)
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo ${args}

    touch ${prefix}_pydamage_summarised.csv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        summarisepydamage: \$(summarise_pydamage.py --version | cut -d ' ' -f 2 )
    END_VERSIONS
    """
}
