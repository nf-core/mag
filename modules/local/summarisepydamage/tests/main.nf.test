// nf-core modules test summarisepydamage
nextflow_process {

    name "Test Process SUMMARISEPYDAMAGE"
    script "../main.nf"
    process "SUMMARISEPYDAMAGE"

    tag "modules"
    tag "modules_"
    tag "summarisepydamage"
    tag "pydamage_analyze"

    setup {
        run('PYDAMAGE_ANALYZE') {
                script '../../../nf-core/pydamage/analyze/main.nf'
                process {
                    """
                    input[0] = [
                        [ id:'test' ],
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                        file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/bam/test.paired_end.sorted.bam.bai', checkIfExists: true)
                    ]
                    """
                }
        }
    }

    test("homo_sapiens - csv") {

        when {
            process {
                """
                input[0] = PYDAMAGE_ANALYZE.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("homo_sapiens - csv - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = PYDAMAGE_ANALYZE.out.csv
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
