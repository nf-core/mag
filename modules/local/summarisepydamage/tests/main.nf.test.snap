{
    "homo_sapiens - csv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pydamage_summarised.csv:md5,d497b394b32609778351527fc7fe16d2"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8bf54a3cd6f2078212d496790c573191"
                ],
                "summary_csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pydamage_summarised.csv:md5,d497b394b32609778351527fc7fe16d2"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8bf54a3cd6f2078212d496790c573191"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-19T13:53:19.753209317"
    },
    "homo_sapiens - csv - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pydamage_summarised.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,8bf54a3cd6f2078212d496790c573191"
                ],
                "summary_csv": [
                    [
                        {
                            "id": "test"
                        },
                        "test_pydamage_summarised.csv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,8bf54a3cd6f2078212d496790c573191"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-08-19T13:53:32.312662338"
    }
}